lib/tasks/migrate_user_organisation_to_user_permission.rake

```diff
@@ -0,0 +1,19 @@
+namespace :organisation_user_migrate do
+  task migrate_organisation_user_to_user_permission: :environment do
+    total_bm = Benchmark.measure do
+      UserPermission.insert_all(
+        OrganisationUser.includes(:user, organisation: [providers: [:recruitment_cycle]]).all.flat_map { |organisation_user|
+          organisation_user.organisation.providers.where(recruitment_cycle: RecruitmentCycle.current).ids.map do |id|
```

Comment:
Do we want to scope this by recruitment cycle? By doing this, we're essentially removing user access to providers from previous recruitment cycles. Does that have any unintended side effects? Does it remove our ability to develop new functionality, eg—giving users a read-only view of previous recruitment cycles? Maybe it would be safer to preserve all existing permissions for now?

cc: @defong @darokel @dankmitchell

Reply:
Thats a good point, I don't see any reason why we shouldn't keep all the existing records. I'll remove the `recruitment_cycle` from the `where`.

Reply:
this looks like it has been resolved